command/kv_put.go

```diff
@@ -161,5 +161,15 @@ func (c *KVPutCommand) Run(args []string) int {
 		return PrintRawField(c.UI, secret, c.flagField)
 	}
 
-	return OutputSecret(c.UI, secret)
+	if Format(c.UI) != "table" {
+		return OutputSecret(c.UI, secret)
+	}
+
+	outputPath(c.UI, path, "Secret Path")
+
+	metadata := secret.Data
+	c.UI.Info(getHeaderForMap("Metadata", metadata))
+	OutputData(c.UI, metadata)
+
+	return 0
```

Comment:
For the `OutputData` call, if I understand it correctly I think we might swallow an error code here. To @averche point you might want to introduce something like:
```go
	if Format(c.UI) == "table" {
		metadata := secret.Data
		c.UI.Info(getHeaderForMap("Metadata", metadata))
		return OutputData(c.UI, metadata)
	}

	return OutputSecret(c.UI, secret)
```

Reply:
Yeah I noticed that throughout the KV code they're actually doing this (calling OutputData without the return in front) so that's where I got that (look at the current kv get code, for example), so I figured there was just some "how to write CLI code" thing I wasn't understanding, but I agree that it seems kinda janky so maybe I will switch it to be like that.